app/controllers/users_controller.rb

```diff
@@ -213,9 +213,19 @@ def change_language(user_data)
   def change_email(user_data)
     @user.unconfirmed_email = user_data[:email]
     if @user.save
-      @user.send_confirm_email
-      if @user.unconfirmed_email
-        flash.now[:notice] = t("users.update.unconfirmed_email_changed")
+      if AppConfig.mail.enable?
+        @user.send_confirm_email
+        if @user.unconfirmed_email
+          flash.now[:notice] = t("users.update.unconfirmed_email_changed")
+        end
+      else
+        @user.email = @user.unconfirmed_email
```

Comment:
We don't need to save the email to `@user.unconfirmed_email` if mail is disabled on that pod. We can save it directly to `@user.email` then.